tests/test_helpers/test_async_utils.py

```diff
@@ -254,3 +245,150 @@ def test_libp2pconnection_awaitable_proc_cancelled():
     proc = AwaitableProc(["sleep", "100"], shell=False)
     proc_task = asyncio.ensure_future(proc.start())
     proc_task.cancel()
+
+
+class RunAndExit(Runnable):
+    """Test class."""
+
+    async def run(self):
+        """Test method."""
+        await asyncio.sleep(0.2)
+
+
+class TestRunnable:
+    """Tests for Runnable object."""
+
+    def test_no_loop_and_threded(self):
+        """Test runnable fails on threaded mode and loop provided.."""
+        with pytest.raises(ValueError,):
+            RunAndExit(loop=asyncio.get_event_loop(), threaded=True)
+
+    @pytest.mark.asyncio
+    async def test_runnable_async(self):
+        """Test runnable async methods."""
+        # for pydocstyle
+        class TestRun(Runnable):
+            async def run(self):
+                while True:
+                    await asyncio.sleep(1)
+
+        run = TestRun()
+        run.start()
+        run.stop()
+        await run.wait_completed()
+
+        run = TestRun(threaded=True)
+        run.start()
+        run.stop()
+        run.wait_completed(sync=True)
+
+        run = RunAndExit()
+        await run.start_and_wait_completed()
+
+    def test_runnable_sync(self):
+        """Test runnable sync methods."""
+        run = RunAndExit()
+        run.start_and_wait_completed(sync=True)
+
+    @pytest.mark.asyncio
+    async def test_double_start(self):
+        """Test runnable async methods."""
+        # for pydocstyle
+        class TestRun(Runnable):
+            async def run(self):
+                while True:
+                    await asyncio.sleep(1)
+
+        run = TestRun()
+        await run.wait_completed()
+        assert run.start()
+        assert not run.start()
+        run.stop()
+        await run.wait_completed()
+        await run.wait_completed()
+
+    @pytest.mark.asyncio
+    async def test_run_in_thread(self):
+        """Test runnable async methods."""
```

Comment:
Docstrings for tests do not match test method names

Reply:
+